src/main/java/jek/collections/IntArrayList.java

```diff
@@ -293,4 +328,20 @@ public boolean containsAll(Collection c) {
     public Object[] toArray(Object[] a) {
         return new Object[0];
     }
+
+    private void increaseArray() {
+        if(end == ial.length) {
+            Object[] ial2 = new Object[ial.length*2];
+            System.arraycopy(ial, 0, ial2, 0, ial.length);
+            ial = ial2;
+        }
+    }
+
+    private void decreaseArray() {
+        if (end < (ial.length/2-1) && end > 10){
```

Comment:
Why 10?

Reply:
10 was our initial size, so I decided not to step over it

Reply:
changed to 9, if you meant this

Reply:
not really... the 10(or 9 in this case already)  is what called "magic number". That's one of types of problems which are identified during the code review. Usually 1 might be there if we're expecting only single entrance. 2 is also ok as you used to increase twice the size of initial array. Usually we move the magic numbers to constants. e.g. INITIAL_ARRAY_SIZE or MULTIPLY_FACTOR or whatever would be suitable.